kpi/tests/api/v2/test_api_asset_snapshots.py

```diff
@@ -239,6 +240,23 @@ def test_xml_renderer_with_invalid_asset(self):
         # Check that the error message contains the expected substring
         self.assertIn('_fail', str(context.exception))
 
+    def test_anonymous_can_create_snapshot_when_asset_shared_public(self):
+        self.client.login(username='someuser', password='someuser')
+        asset = self.create_asset('Public asset', self.form_source, format='json')
+        asset_url = reverse(self._get_endpoint('asset-detail'), args=(asset.uid,))
+
+        # Assign 'view' permission on the asset to AnonymousUser
+        anon_user = AnonymousUser()
+        self.add_perm(asset, anon_user, 'view_')
```

Comment:
I was not aware about this method, we should get rid of it entirely.
Let's use `asset.assign_perm(get_anonymous_user(), PERM_VIEW_ASSET)` instead (for new code).